### PR TITLE
feat: allow users to set styles on icons in Vue

### LINKED ISSUE #4785: User cannot set dynamic styles on an icon in Vue template.
Ensure dynamic classes work when icon in @carbon/icons-vue is not given a static class.

## What package(s) are you using?

- [x] @carbon/icons-vue
 
## Detailed description

> Describe in detail the issue you're having.
It was not possible to specify only dynamic classes to @carbon/icons-vue

> Is this issue related to a specific component?
All icons generated by @carbon-icons/vue

> What did you expect to happen? What happened instead? What would you like to
> see changed?
To be able to specify dynamic styles on a Carbon Icon in Vue template

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
10.7

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
@caronb/vue

## Steps to reproduce the issue

PR with test to follow.

### PR DESCRIPTION
**Pull request by lee-chase**
Closes #4785

Allows user to specify styles dynamically in a Vue template, including overriding individual styles such as 'will-change' where set by default in getAttributes.

#### Changelog

M       packages/icons-vue/tasks/__tests__/createIconComponent-test.js
M       packages/icons-vue/tasks/createIconComponent.js

### CODE REVIEW

**Review by emyarod — COMMENTED** [review]
I believe the merge conflict is due to your previous PR, would you mind resolving that conflict?

**Comment by lee-chase** [comment]
> I believe the merge conflict is due to your previous PR, would you mind resolving that conflict?

Merged

**Review by joshblack — APPROVED** [review]

File: packages/icons-vue/tasks/createIconComponent.js [hunk 1]
```diff
@@ -58,6 +59,23 @@ export default {
...
       svgData.class[data.class] = true;
     }
+
+    // combine incoming staticStyle, style and internal style attribute.
+    const style = svgData.attrs.style;
+    delete svgData.attrs.style;
+
+    // convert incoming style attr to object form.
+    const styleArray = style.split(';').filter(item => item.trim().length > 0);
+    const styleObj = styleArray.reduce((acc, styleString) => {
```
Comment (joshblack) [inline packages/icons-vue/tasks/createIconComponent.js]: Would you want to do this at build time versus compile time? In our base component for React we end up just building the object in scope:

https://github.com/carbon-design-system/carbon/blob/master/packages/icon-build-helpers/src/builders/react/components/Icon.js#L12-L14

This value used to derive from the value coming from icon-helpers (e.g. we parsed the string value) but now we just inline it to make things easier.

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by dakahn — APPROVED** [review]
do the bit of refactoring @joshblack  mentioned and this looks good to me 👍

File: packages/icons-vue/tasks/createIconComponent.js (hunk 1 above)
Comment (lee-chase) [reply packages/icons-vue/tasks/createIconComponent.js]: Is this what you were suggesting @joshblack ?

**Review by joshblack — APPROVED** [review]

**Pull request merged**